CMakeLists.txt

```diff
@@ -795,17 +795,24 @@ set (Seastar_PRIVATE_CXX_FLAGS
   -UNDEBUG
   -Wall
   -Werror
-  -Wimplicit-fallthrough
-  -Wdeprecated
-  -Wno-error=deprecated)
+  -Wimplicit-fallthrough)
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
   include (CheckGcc107852)
   if (NOT Cxx_Compiler_BZ107852_Free)
     list (APPEND Seastar_PRIVATE_CXX_FLAGS
       -Wno-error=stringop-overflow
       -Wno-error=array-bounds)
-  endif()
+  endif ()
+  list (APPEND Seastar_PRIVATE_CXX_FLAGS
+    -Wdeprecated-declarations
+    -Wno-error=deprecated-declarations)
+endif ()
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  list (APPEND Seastar_PRIVATE_CXX_FLAGS
+    -Wdeprecated
+    -Wno-error=deprecated)
```

Comment:
Shouldn't this be unconditional? We do want to warn about using deprecated features.

Reply:
I think it's undocumented because its behaviour is already described by `-Wno-deprecated` – if that option disables warnings about using deprecated features, `-Wdeprecated` makes the compiler warn about them. The manual says this:

> Many options have long names starting with `-f` or with `-W`---for example, `-fmove-loop-invariants`, `-Wformat` and so on.  Most of these have both positive and negative forms; the negative form of `-ffoo` is `-fno-foo`. This manual documents only one of these two forms, whichever one is not the default.

while [the documentation](https://gcc.gnu.org/onlinedocs/gcc-13.2.0/gcc/Warning-Options.html) this:

> You can request many specific warnings with options beginning with ‘`-W`’, for example `-Wimplicit` to request warnings on implicit declarations. Each of these specific warning options also has a negative form beginning ‘`-Wno-`’ to turn off warnings; for example, `-Wno-implicit`. This manual lists only one of the two forms, whichever is not the default.

I also think enabling those options unconditionally would be a good idea, or at least `-Wno-error=deprecated` as `-Wdeprecated` is enabled by default.